fairtools/MCConfigurator/FairXmlVMCConfig.cxx

```diff
@@ -0,0 +1,429 @@
+            LOG(debug) << "Setting Process \"" << fXmlEngine->GetNodeName(child) << "\" to " << ConvertTo<float>(child);
+            child = fXmlEngine->GetNext(child);
+        }
+    }
+
+    if ((xmlNode = GetPointer("SIM:MonteCarlo:Cut", fXmlConfig))) {
+        TVirtualMC* MC = TVirtualMC::GetMC();
+        XMLNodePointer_t child = fXmlEngine->GetChild(xmlNode);
+        while (child != 0) {
+            fXmlEngine->GetNodeName(child);
+            MC->SetCut(fXmlEngine->GetNodeName(child), ConvertTo<float>(child));
+            LOG(debug) << "Setting Cut \"" << fXmlEngine->GetNodeName(child) << "\" to " << ConvertTo<float>(child);
+            child = fXmlEngine->GetNext(child);
+        }
+    }
+}
+
+string FairXmlVMCConfig::ObtainXmlFileName(const char* mcEngine)
+{
+    TString lUserConfig = FairRunSim::Instance()->GetUserConfig();
+
+    TString work = getenv("VMCWORKDIR");
+    TString work_config = work + "/gconfig/";
+    work_config.ReplaceAll("//", "/");
+
+    TString config_dir = getenv("CONFIG_DIR");
+    config_dir.ReplaceAll("//", "/");
+
+    Bool_t AbsPath = kFALSE;
+
+    TString configFileWithPath;
+
+    TString configFile;
+    if (lUserConfig.IsNull()) {
+        if (strcmp(mcEngine, "TGeant4") == 0) {
+            configFile = "g4Config.xml";
+        }
+        if (strcmp(mcEngine, "TGeant3") == 0) {
+            configFile = "g3Config.xml";
+        }
+        lUserConfig = configFile;
+    } else {
+        if (lUserConfig.Contains("/")) {
+            AbsPath = kTRUE;
+        }
+        configFile = lUserConfig;
+        LOG(debug) << "---------------User config is used: " << configFile.Data();
+    }
+    if (!AbsPath && TString(gSystem->FindFile(config_dir.Data(), configFile)) != TString("")) {
+        LOG(debug) << "---------------CONFIG_DIR is used: " << config_dir.Data() << " to get \"" << configFile.Data()
+                   << "\"";
+        configFileWithPath = configFile;
+    } else {
+        if (AbsPath) {
+            configFileWithPath = lUserConfig;
+        } else {
+            configFileWithPath = work_config + lUserConfig;
+        }
+    }
+    return configFileWithPath.Data();
+}
+
+void FairXmlVMCConfig::StoreXmlInfo()
+{
+    std::ostringstream nodestring;
+    nodestring << fMCEngine << "\n";
+    nodestring << fXmlConfig;
+    nodestring << "\n";
+    TObjString* configObject = new TObjString(nodestring.str().c_str());
+    FairRunSim::Instance()->GetSink()->WriteObject(configObject, "SimulationSetup", TObject::kSingleKey);
+
+    LOG(debug) << "FairXmlVMCConfig::StoreXmlInfo() done.";
+}
+
+XMLNodePointer_t FairXmlVMCConfig::GetPointer(string path, XMLNodePointer_t node, int verbose)
+{
+    // this function display all accessible information about xml node and its children
+    if (verbose)
+        cout << "GetPointer with path \"" << path << "\"" << endl;
+    path.replace(0, path.find(':') + 1, "");
```

Comment:
It immediately discards the first element from the `path`? Meaning it just accepts any root element?

Reply:
`path.find(':') + 1` this also deserves a comment IMHO that the integer overflow is intended (if it is).